HW_8/task_2.py

```diff
@@ -0,0 +1,46 @@
+import string
+
+
+def is_valid_pass(password):
+    kol_lowercase = 0
+    kol_uppercase = 0
+    kol_digit = 0
+    kol_character = 0
+    for i in password:
+        if i in string.ascii_lowercase:
+            kol_lowercase += 1
+        if i in string.ascii_uppercase:
+            kol_uppercase += 1
+        if i.isdigit() == True:
+            kol_digit += 1
+        if i in "$#@":
+            kol_character += 1
+
+    if kol_lowercase == 0 or kol_uppercase == 0 or kol_digit == 0 or kol_character == 0:
+        return False
+
+    if len(password) < 6 or len(password) > 16:
+        return False
+
+    return True
+
+
+def is_valid_pass_v2(password):
+    if sum(map(str.islower, password)) == 0:
+        return False
+    if sum(map(str.isupper, password)) == 0:
+        return False
+    if sum(map(str.isdigit, password)) == 0:
+        return False
+    if "@" not in password and "#" not in password and "$" not in password:
+        return False
+    if len(password) < 6 or len(password) > 16:
+        return False
+    return True
+
+
+p = input()
+if is_valid_pass_v2(p) == True:
+    print("All OK")
+else:
+    print("Error")
```

Comment:
another ways

import re

def validate_password(input_password):
    
    flag = True
    
    while flag:  
        if len(input_password)<6 or len(input_password)>16:
            break
        elif not re.search("[a-z]",input_password):
            break
        elif not re.search("[0-9]",input_password):
            break
        elif not re.search("[A-Z]",input_password):
            break
        elif not re.search("[$#@]",input_password):
            break
        # Returns a match where the string contains a white space character
        elif re.search("\s",input_password):  
            break
        else:
            return f"{input_password} is valid password"
            flag = False
            break

    if flag:
        return f"{input_password} is invalid password"


print(validate_password("ER45jknkjbyhdfy"))
print(validate_password("ER45@hjkyutt TYTIU566"))
print(validate_password("DFG567#oijl89"))

##################################

import re

password = input("Enter your password:\n")

if re.match("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[$#@])(?=.{6,16}$).*", password):
    print("Nice pass! Psss, I won't tell anyone ;)")
else:
    print("Password is not valid. Check if you have:"
          """
    At least 1 letter between[a-z] and 1 letter between[A-Z].
    At least 1 number between[0-9].
    At least 1 character from [$#@].
    Minimum length 6 characters.
    Maximum length 16 characters.
          """
           )